packages/core/src/di/interface/defs.ts

```diff
@@ -175,12 +175,32 @@ export function ɵɵdefineInjector(options: {factory: () => any, providers?: any
 }
 
 /**
- * Read the `ngInjectableDef` type in a way which is immune to accidentally reading inherited value.
+ * Read the `ngInjectableDef` for `type` in a way which is immune to accidentally reading inherited
+ * value.
  *
- * @param type type which may have `ngInjectableDef`
+ * @param type A type which may have its own (non-inherited) `ngInjectableDef`.
  */
 export function getInjectableDef<T>(type: any): ɵɵInjectableDef<T>|null {
-  return type && type.hasOwnProperty(NG_INJECTABLE_DEF) ? (type as any)[NG_INJECTABLE_DEF] : null;
+  return type && type.hasOwnProperty(NG_INJECTABLE_DEF) ? type[NG_INJECTABLE_DEF] : null;
+}
+
+/**
+ * Read the `ngInjectableDef` for`type` or its ancestors.
+ *
+ * @param type A type which may have `ngInjectableDef`, via inheritance.
+ *
+ * @deprecated Will be removed in v9.
```

Comment:
I think we decided v10, no? 

```suggestion
 * @deprecated Will be removed in v10.
```

Reply:
I thought that if we could deprecate it in v8 then we could kill it in v9? But happy to go with v10 for safety.

Reply:
If the intent is to reduce dev pain, I think v10 is better because it will give people more time to migrate. Otherwise, it will break them in v9 with other Ivy changes